domain/modelupgrade/params.go

```diff
@@ -0,0 +1,17 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package modelupgrade
+
+import (
+	"github.com/juju/juju/core/semversion"
+)
+
+// UpgradeModelParams holds the parameters used to upgrade a model.
+type UpgradeModelParams struct {
+	ControllerModelVersion semversion.Number
+	TargetVersion          semversion.Number
```

Comment:
This member has a slight smell to me. It is both being used a version value and that of a bool to control logic. i.e if this value is zero we follow the controller version.

My suggestion would be that the service needs two methods:
- UpgradeModelToLatest
- UpgradeModelToVersion

Reply:
The code is a 1:1 copy / port of what we have in the facade. I was trying to avoid rewriting the logic as part of the refactor which is simply a relocation of the code so that we can access the provider via the service. But can look at any reasonable improvements.